File: C-C++/node-locked-license/node-locked-license.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include "../header/LexActivator.h"

using namespace std;

// Uncomment following for Windows static build
/*
#if _WIN32
#define LEXACTIVATOR_STATIC
#pragma comment(lib, "winhttp")
#if _WIN64
#pragma comment(lib, "x64/libcurl_MD")
#pragma comment(lib, "x64/LexActivator")
#else
#pragma comment(lib, "x86/libcurl_MD")
#pragma comment(lib, "x86/LexActivator")
#endif
#endif
*/

#if _WIN32
#if _WIN64
#pragma comment(lib, "x64/LexActivator")
#else
#pragma comment(lib, "x86/LexActivator")
#endif
#endif

void init()
{
	int status;

	/*
	 * Product data needs to be hard-coded and this value is not sensitive or secret.
	 */

#if _WIN32
	status = SetProductData(L"PASTE_CONTENT_OF_PRODUCT.DAT_FILE");
#else
	status = SetProductData("PASTE_CONTENT_OF_PRODUCT.DAT_FILE");
#endif
	if (LA_OK != status)
	{
		printf("Error code: %d", status);
		getchar();
		exit(status);
	}

	/*
	 * Product Id is simply a unique identifier of your product and needs to be hard-coded.
	 * Similarly this Id is not sensitive as well.
	 */

#if _WIN32
	status = SetProductId(L"PASTE_PRODUCT_ID", LA_USER);
#else
	status = SetProductId("PASTE_PRODUCT_ID", LA_USER);
#endif
	if (LA_OK != status)
	{
		printf("Error code: %d", status);
		getchar();
		exit(status);
	}
}

/*
 * Ideally on a button click inside a dialog
 */

void activate()
{
	int status;
#if _WIN32
	status = SetLicenseKey(L"PASTE_LICENCE_KEY");
#else
	status = SetLicenseKey("PASTE_LICENCE_KEY");
#endif
	if (LA_OK != status)
	{
		printf("Error code: %d", status);
		getchar();
		exit(status);
	}

	status = ActivateLicense();
	if (LA_OK == status || LA_EXPIRED == status || LA_SUSPENDED == status)
	{
		printf("License activated successfully: %d", status);
	}
	else
	{
		printf("License activation failed: %d", status);
	}
}

int main()
{
	init();

	activate(); // in case of node-locekd this function only needs to be called once.

	int status = IsLicenseGenuine();
	if (LA_OK == status)
	{
		// You can now run your code as the user is genuine/licensed.

		unsigned int expiryDate = 0;
		GetLicenseExpiryDate(&expiryDate);
		int daysLeft = (expiryDate - time(NULL)) / 86400;
		printf("Days left: %d\n", daysLeft);
		printf("License is genuinely activated!");
	}
	else if (LA_EXPIRED == status)
	{
		printf("License is genuinely activated but has expired!");
	}
	else if (LA_SUSPENDED == status)
	{
		printf("License is genuinely activated but has been suspended!");
	}
	else if (LA_GRACE_PERIOD_OVER == status)
	{
		printf("License is genuinely activated but grace period is over!");
	}
	else
	{
		printf("Something went wrong! Error code: %d", status);
	}
	getchar();
	return 0;
}

File: NodeJs/node-locked-license/node-locked-license.js
/* eslint-disable new-cap */
import { LexActivator, LexStatusCodes, LexActivatorException, PermissionFlags } from '@cryptlex/lexactivator'

function init() {
    /*
     * Product data needs to be hard-coded and this value is not sensitive or secret.
     * You can download the product.dat file from the product page of the admin dashboard.
     */
    LexActivator.SetProductData('PASTE_CONTENT_OF_PRODUCT.DAT_FILE');

    /*
     * Product Id is simply a unique identifier of your product and needs to be hard-coded.
     * Similarly this Id is not sensitive as well.
     */

    LexActivator.SetProductId('PASTE_PRODUCT_ID', PermissionFlags.LA_USER);
}

// Ideally on a button click inside a dialog.

function activate() {
	LexActivator.SetLicenseKey('PASTE_LICENCE_KEY');
    /*
     * SetActivationMetadata() can be used to get info from client to server.
     * The metadata appears along with the activation details of the license on the dashboard.
     * This function is not a required function.
     */
	LexActivator.SetActivationMetadata('key1', 'value1');
	const status = LexActivator.ActivateLicense();
	if (LexStatusCodes.LA_OK == status || LexStatusCodes.LA_EXPIRED == status || LexStatusCodes.LA_SUSPENDED == status) {
		console.log('License activated successfully:', status);
	} else {
		console.log('License activation failed:', status);
	}
}

function licenseCallback(status) {
    try {
        switch (status) {
            case LexStatusCodes.LA_OK:
            case LexStatusCodes.LA_EXPIRED:
            case LexStatusCodes.LA_SUSPENDED:
                console.log('License is genuinely activated!', status);
                break;
            default:
                throw new LexActivatorException(status);
        }
    } catch (error) {
        console.log(error.code, error.message);
    }
}

function main() {
    try {
        init();
        activate(); // in case of node-locekd this function only needs to be called once.

        /*
         * Whenever the server sync occurs in a separate thread, and server returns the response,
           license callback function gets invoked.
         */
        LexActivator.SetLicenseCallback(licenseCallback);

        const status = LexActivator.IsLicenseGenuine();
        if (LexStatusCodes.LA_OK == status) {
            console.log('License is genuinely activated!');

            // You can execute your code as the user is licensed

            const expiryDate = LexActivator.GetLicenseExpiryDate();
            const daysLeft = (expiryDate - (new Date().getTime() / 1000)) / 86400;
            console.log('Days left:', daysLeft);
            
        }
        else if (LexStatusCodes.LA_EXPIRED == status) {
            console.log('License is genuinely activated but has expired!');
        } else if (LexStatusCodes.LA_SUSPENDED == status) {
            console.log('License is genuinely activated but has been suspended!');
        } else if (LexStatusCodes.LA_GRACE_PERIOD_OVER == status) {
            console.log('License is genuinely activated but grace period is over!');
        }

    } catch (error) {
        console.log(error.code, error.message);
    }
}

main();